app.rb

```diff
@@ -0,0 +1,181 @@
+      options
+    end
+  end
+
+  def menu
+    puts
+    p 'School Library App!'
+    puts
+    option = nil
+    while option != 7
+      puts 'Please choose an option by entrering a number: '
+      options
+      print '[Input number]: '
+      option = gets.chomp.strip.to_i
+      check(option)
+      puts
+    end
+    puts 'GOOD BYE'
+  end
+
+  def list_books
+    key = 0
+    puts
+    puts 'Books'
+    puts 'No book available! Choose (4) to  create one' if @books.empty?
+    @books.each do |book|
+      puts "#{key}- Title: #{book.title} Author: #{book.author}"
+      key += 1
+    end
+  end
+
+  def list_people
+    key = 0
+    puts
+    puts 'People'
+    puts 'No people available! Choose (3) to create a person ' if @people.empty?
+    @people.each do |person|
+      print "#{key}- [#{person.class.name}] ID: #{person.id} Name: #{person.name} Age: #{person.age} "
+      print "Parent Permission: #{person.parent_permission}" if person.is_a?(Student)
+      print "Specialisation: #{person.specialization}" if person.is_a?(Teacher)
+      puts
+      key += 1
+    end
+  end
+
+  def list_rentals
+    puts
+    fetch_people
+    puts 'Choose Person ID?: '
+    entry = gets.chomp.to_i
+    puts 'Rentals'
+    @people.each do |person|
+      next unless person.id == entry
+
+      person.rentals.each do |rental|
+        puts "Date: #{rental.date} Book title: #{rental.book.title} by #{rental.book.author}"
+        puts
+      end
+    end
+  end
+
+  def create_person
+    entry = nil
+    print 'Choose (1) to Create a Student or (2) for a Teacher: '
+    until [1, 2].include?(entry)
+      entry = gets.chomp.strip.to_i
+      puts 'Choice is not correct : (1) for Student, (2) for Teacher' unless [1, 2].include?(entry)
+    end
+    case entry
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      p 'Choose between 1 and 2'
+    end
+  end
+
+  def create_teacher
+    age = -1
+    print 'Age: '
+    while age <= 0
+      age = gets.chomp.to_i
+      print 'Add correct age of teacher: ' if age <= 0
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip.capitalize
+    @people.push(Teacher.new(age, specialization, name))
+    puts 'Teacher created successfuly!'
+  end
+
+  def create_student
+    print 'Age: '
+    age = -1
+    while age <= 0
+      age = gets.chomp.to_i
+      print 'Add correcte age of student: ' if age <= 0
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has parent permission ? (N/Y): '
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'Y', 'YES'
+      permission = true
+    when 'N', 'NO'
+      permission = false
+    end
+    @people.push(Student.new(nil, age, name, parent_permission: permission))
+    puts 'Student created successfuly!'
+  end
+
+  def create_rental
+    fetch_books
+    print 'Select the key of the book: '
```

Comment:
- [ ] Your app.rb file that will serve as their console app entry-point should have methods that do and look something like the following:


```ruby
# *Create a person (teacher or student, not a plain `Person`)
  def create_person
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    # *If needed, ask for parameters for the option.
    person_type = gets.chomp
    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end

    print "Age: "
    # *If needed, ask for parameters for the option.
    age = gets.chomp

    print "Name: "
    # *If needed, ask for parameters for the option.
    name = gets.chomp

    person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
      end

    @people << person
    puts "Person created successfully"
  end

  # *Create a book.
  def create_book
    print "Title: "
    # *If needed, ask for parameters for the option.
    title = gets.chomp

    print "Author: "
    # *If needed, ask for parameters for the option.
    author = gets.chomp

    @books << Book.new(title, author)
    puts "Book created successfully"
  end
  
  # *Create a rental.
  def create_rental
    puts "Select a book from the following list by number"
    @books.each_with_index do |book, index|
      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
    end
    # *If needed, ask for parameters for the option.
    book_index = gets.chomp.to_i

    puts "\nSelect a person from the following list by number (not id)"
    @people.each_with_index do |person, index|
      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
    end
    # *If needed, ask for parameters for the option.
    person_index = gets.chomp.to_i

    print "\nDate: "
    # *If needed, ask for parameters for the option.
    date = gets.chomp

    @rentals << Rental.new(date, @books[book_index], @people[person_index])
    puts "Rental created successfully"
  end
  
```